client/src/components/articles/ArticleComment.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div v-if="isCurrentCommentVisible" class="mt-5 flex">
+    <div class="flex phone:mr-6 h-20 w-3/12 phone:w-full">
+      <img class="h-16 w-16 rounded-full border-2 border-red-600" :src="getUserPicture" />
+    </div>
+    <div class="pb-4 pr-2 text-left">
+      <p class="text-lg font-semibold text-black">{{ user.username }}</p>
+      <p class="text-sm text-slate-500">{{ formattedDate }}</p>
+      <p class="mt-4 text-lg text-black">{{ comment.body }}</p>
+    </div>
+    <CommentIcons v-if="isUserTheAuthor" @toggleComment="toggleComment" @deleteComment="deleteComment" />
+  </div>
+  <div class="flex justify-between">
+    <button
+    v-if="getUserId"
+      class="flex h-16 w-28 rounded-xl transition-transform duration-300 ease-in-out hover:scale-125 hover:cursor-pointer phone:ml-2 phone:h-14 phone:w-18"
+      :class="getLikedClass"
+      @click.prevent="liked"
+    >
+      <LikeIcon />
+      <p class="ml-1 mt-2">{{ updatedLikes }}</p>
+    </button>
+    <button
+      v-if="!comment.parentId && isCurrentCommentVisible && getUserId"
+      class="text-md h-12 w-24 bg-gray-300 text-black hover:cursor-pointer"
+      @click="toggleComment('Reply')"
+    >
+      Reply
+    </button>
+  </div>
+  <div class="mb-8 ml-28 mt-8 w-10/12">
+    <CreateComment
+      v-if="isReplyCommentVisible"
+      :user="user"
+      :comment="category"
+      @cancelComment="toggleComments"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useCommentStore } from '../../stores/CommentStore'
+import { useUserStore } from '../../stores/UserStore'
+import CreateComment from './CreateComment.vue'
+import LikeIcon from '../icons/LikeIcon.vue'
+import CommentIcons from '../icons/CommentIcons.vue'
+import { formatDate } from '../../helpers/helper.ts'
+import { User } from '../../types/user.ts'
+import { Comment } from '../../types/comment.ts'
+
+const { setSingleComment, likedComment, deleteToComment } = useCommentStore()
```

Comment:
Most of the time we are not destructuring the store, because without it, we can differentiate easier which is a store function and which is a component function. Both is correct but without destructuring we can aviod functions like this

```
function deleteComment() {
  deleteToComment(props.comment)
}
```